webapp-client/src/component/AddCategoryForm.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+
+export default class AddCategoryForm extends React.Component {
+  /** This function will be changed later to directly POST request to db instead of calling addCategory */
+  createCategory = (e) => {
+    e.preventDefault();
+    const category = this._inputElement.value;
+    this.props.addCategory(category);
+    this._inputElement.value = '';
+  };
+  render() {
+    return (
+      <form className="my-1" onSubmit={this.createCategory}>
+        <input
+          ref={(input) => (this._inputElement = input)}
+          type="text"
+          className="input is-rounded"
+          placeholder="New Category"
```

Comment:
You generally want to avoid omitting a label and only using a placeholder for an input. Some articles about this:

- https://www.nngroup.com/articles/form-design-placeholders/
- https://www.smashingmagazine.com/2018/06/placeholder-attribute/
- https://medium.theuxblog.com/4-reasons-to-avoid-using-placeholder-text-in-forms-8fc33e724e4f


Optional: Then you can shorten the button text to say "Add".